### PR TITLE
Capture point inequality

### PR DESCRIPTION
**Pull request by PedroDesRobots**
Hi,

I have added the capture point inequality based on the paper of : 

Ramos, O. E., Mansard, N., & Soueres, P. (2014). Whole-body Motion Integrating the Capture Point in the Operational Space Inverse Dynamics Control. In IEEE-RAS International Conference on Humanoid Robots (Humanoids).

At the moment this class handles only rectangular shape concerning the support polygon.

Let me know what you think and how to improve it.

### CODE REVIEW

File: src/tasks/task-capture-point-inequality.cpp [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        b_upper(i) = m_ka*(m_rp_max(i) - m_p_com(i) - m_v_com(i)*(m_delta_t+1/m_w));
+      }
+
+      m_constraint.lowerBound() =  b_lower - m_drift.head(m_dim); 
+      m_constraint.upperBound() =  b_upper - m_drift.head(m_dim);
+ 
+      m_constraint.setMatrix(Jcom.block(0,0,m_dim,m_nv));
+
+      b_lower.setZero(m_dim);
+      b_upper.setZero(m_dim);
```
Comment (andreadelprete) [inline src/tasks/task-capture-point-inequality.cpp:132]: Why do you need to reset b_lower and b_upper here?

File: src/tasks/task-joint-posVelAcc-bounds.cpp [hunk 2]
```diff
@@ -17,7 +17,10 @@
 
 #include <tsid/tasks/task-joint-posVelAcc-bounds.hpp>
 #include "tsid/robots/robot-wrapper.hpp"
+<<<<<<< HEAD
+=======
 // #include <tsid/utils/stop-watch.hpp>
+>>>>>>> ebc2966e0186cc3b354ca04912dd7c9b4dc198bc
```
Comment (andreadelprete) [inline src/tasks/task-joint-posVelAcc-bounds.cpp:23]: Also in this file you have unresolved conflicts

File: src/tasks/task-capture-point-inequality.cpp [hunk 3]
```diff
@@ -0,0 +1,138 @@
+//
+// Copyright (c) 2017 CNRS, NYU, MPI Tübingen
```
Comment (andreadelprete) [inline src/tasks/task-capture-point-inequality.cpp:2]: You can change the year and add PAL Robotics here

File: CMakeLists.txt [hunk 4]
```diff
@@ -79,6 +79,7 @@ SET(BOOST_COMPONENTS filesystem system unit_test_framework)
 
 IF(BUILD_PYTHON_INTERFACE)
   FINDPYTHON()
+  INCLUDE_DIRECTORIES(SYSTEM ${PYTHON_INCLUDE_DIRS})
```
Comment (andreadelprete) [inline CMakeLists.txt:82]: Why did you add this?

File: CMakeLists.txt [hunk 5]
```diff
@@ -28,7 +28,7 @@ OPTION(INSTALL_PYTHON_INTERFACE_ONLY "Install *ONLY* the python bindings" OFF)
 OPTION(SUFFIX_SO_VERSION "Suffix library name with its version" ON)
 OPTION(INITIALIZE_WITH_NAN "Initialize Eigen entries with NaN" OFF)
 OPTION(EIGEN_RUNTIME_NO_MALLOC "If ON, it can assert in case of runtime allocation" ON)
-OPTION(EIGEN_NO_AUTOMATIC_RESIZING "If ON, it forbids automatic resizing of dynamics arrays and matrices" OFF)
+OPTION(EIGEN_NO_AUTOMATIC_RESIZING "If ON, it forbids automatic resizing of dynamics arrays and matrices" ON)
```
Comment (andreadelprete) [inline CMakeLists.txt:31]: I'd reset this to OFF by default. If somebody wants to turn it on, they can always do it with cmake

File: include/tsid/tasks/task-joint-posVelAcc-bounds.hpp [hunk 6]
```diff
@@ -47,8 +47,12 @@ namespace tsid
 
       TaskJointPosVelAccBounds(const std::string & name,
                       RobotWrapper & robot,
+<<<<<<< HEAD
+                      double dt);
+=======
```
[the remark this answers is not in the document]
Comment (PedroDesRobots) [reply include/tsid/tasks/task-joint-posVelAcc-bounds.hpp]: it's resolved.

File: src/tasks/task-capture-point-inequality.cpp (hunk 3 above)
Comment (PedroDesRobots) [reply src/tasks/task-capture-point-inequality.cpp]: done.

File: CMakeLists.txt (hunk 4 above)
Comment (PedroDesRobots) [reply CMakeLists.txt]: remove, I don't know why it was added.

File: CMakeLists.txt (hunk 5 above)
Comment (PedroDesRobots) [reply CMakeLists.txt]: I put it OFF

**Comment by andreadelprete** [comment]
I've merged it. Thanks again for your contribution!

**Comment by andreadelprete** [comment]
@PedroDesRobots if you have a Python script demonstrating how to use this new task we could add it to the repo as well.

File: src/tasks/task-capture-point-inequality.cpp [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      m_support_limits_x.setZero(m_dim);
+      m_support_limits_y.setZero(m_dim);
+
+      m_rp_max.setZero(m_dim);
+      m_rp_min.setZero(m_dim);
+
+      b_lower.setZero(m_dim);
+      b_upper.setZero(m_dim);
+
+      m_g = 9.81;
```
Comment (nim65s) [inline src/tasks/task-capture-point-inequality.cpp:57]: Maybe we should use `robot.gravity`
